src/EventSubscriber/GenericEventSubscriber.php

```diff
@@ -111,11 +119,17 @@ public function onRulesEvent(Event $event, $event_name) {
       );
     }
 
-    // Loop over all rules and execute them.
-    foreach ($configs as $config) {
-      /** @var \Drupal\rules\Entity\ReactionRuleConfig $config */
-      $config->getExpression()
-        ->executeWithState($state);
+    // Invoke the rules.
+    foreach ($triggered_events as $triggered_event) {
+      // @todo Only load active reaction rules here.
+      $configs = $storage->loadByProperties(['event' => $triggered_event]);
+
+      // Loop over all rules and execute them.
+      foreach ($configs as $config) {
+        /** @var \Drupal\rules\Entity\ReactionRuleConfig $config */
+        $config->getExpression()
+          ->executeWithState($state);
```

Comment:
Why do we want that? Such that added variables in a rule do not interact with another reaction rule? Makes sense, but that should probably be added with some dedicated test-coverage only.